core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/waterfallChart/WaterfallOpDesc.scala

```diff
@@ -0,0 +1,91 @@
+package edu.uci.ics.texera.workflow.operators.visualization.waterfallChart
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle
+import edu.uci.ics.texera.workflow.common.metadata.{OperatorGroupConstants, OperatorInfo}
+import edu.uci.ics.texera.workflow.common.operators.PythonOperatorDescriptor
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType, Schema}
+import edu.uci.ics.amber.engine.common.workflow.{InputPort, OutputPort}
+import edu.uci.ics.texera.workflow.operators.visualization.{
+  VisualizationConstants,
+  VisualizationOperator
+}
+import edu.uci.ics.texera.workflow.common.metadata.annotations.AutofillAttributeName
+
+class WaterfallChartOpDesc extends VisualizationOperator with PythonOperatorDescriptor {
+
+  @JsonProperty(value = "xColumn", required = true)
+  @JsonSchemaTitle("X Axis Values")
+  @JsonPropertyDescription("The column for x axis values")
+  @AutofillAttributeName
+  var xColumn: String = _
+
+  @JsonProperty(value = "yColumn", required = true)
+  @JsonSchemaTitle("Y Axis Values")
+  @JsonPropertyDescription("The column for y axis values")
+  @AutofillAttributeName
+  var yColumn: String = _
+
+  override def getOutputSchema(schemas: Array[Schema]): Schema = {
+    Schema.builder().add(new Attribute("html-content", AttributeType.STRING)).build()
+  }
+
+  override def operatorInfo: OperatorInfo =
+    OperatorInfo(
+      "Waterfall Chart",
+      "Visualize data as a waterfall chart",
+      OperatorGroupConstants.VISUALIZATION_GROUP,
+      inputPorts = List(InputPort()),
+      outputPorts = List(OutputPort())
+    )
+
+  def createPlotlyFigure(): String = {
+    s"""
+       |        x_values = table['$xColumn']
+       |        y_values = table['$yColumn']
+       |
+       |        fig = go.Figure(go.Waterfall(
+       |            name="Waterfall", orientation="v",
+       |            measure=["relative"] * (len(y_values) - 1) + ["total"],
```

Comment:
Is there a specific reason why the last measure is labeled as "total"? I believe all the measures should be consistent (either all "relative" or all "total") since users cannot specify the measure. Having an inconsistent list may confuse users.

Reply:
Just having them all relative would leave the graph with no special focus on a cumulative "total. The last measure is "total" to highlight the cumulative outcome.